zio-http/src/main/scala/zhttp/http/HeaderExtension.scala

```diff
@@ -135,8 +144,3 @@ private[zhttp] trait HeaderExtension[+A] { self =>
   def hasHeader(name: CharSequence): Boolean =
     getHeaderValue(name).nonEmpty
 }
```

Comment:
Why are these changes required?

Reply:
`getHeader` uses `contentEqualsIgnoreCase` which is private for `HeaderExtension` and it has logic for comparing name of the header that is `AsciiString` (e.g. `HttpHeaderNames.ORIGIN`) with `CharSequence` that is in `Header.name`. I didn't want to duplicate code so I created `HeadersList` which extends `ReadHeaderExtension` which has only read-only methods from `HeaderExtension` (including `getHeader` used in CORS middleware).

Reply:
Other options to solve it:
 * use `HeadersList` as some kind of value class, move there all methods that are currently in `HeaderExtension`; in `Request` and `Response` use `HeaderList` as val and delegate methods like `getHeader` to this class
  * instead of trait use implicit class which will add methods to `List[Header]`, that will only work for "read" functions, "write" function would still need a trait like current, but it will be much smaller and "read" methods will be available everywhere on `List[Header]` when import of implicit class will be added.

Reply:
Ok got it. Would you mind duplicating that bit of code for now? I am facing difficulty in understanding what `HeaderList` really stands for and if it should be part of extensions.

Reply:
Sure, I can do this, but I need some time now, I should be available in ~2-3h

Reply:
Done, You can take a look @tusharmath